lane.py

```diff
@@ -1099,78 +1081,58 @@ def _groups_query(self, _db, lanes):
             library.minimum_featured_quality,
             self.uses_customlists
         )
-        qu = self.works(_db, facets=facets)
-
-        # Include a field that assigns a work to one of the sublanes,
-        # or to the parent lane if it doesn't match any of the
-        # sublanes.
-        qu = self._add_lane_id_field(_db, qu, lanes, target_size)
+        first_query = None
+        other_queries = []
+        for lane in lanes:
+            for mw, quality_tier in lane.works_in_window(
+                    _db, facets, target_size
+            ):
+                yield mw, quality_tier, lane
 
-        # We order by quality tier, then by lane, then randomly.  This
-        # ensures that if we have to apply a LIMIT, the LIMIT is more
-        # likely to cut off low-quality results for an early lane than
-        # high-quality results for a late lane.
-        qu = qu.order_by(
-            "quality_tier desc", "lane_id", work_model.random.desc()
-        )
+    def works_in_window(self, _db, facets, target_size):
+        """Find all MaterializedWorkWithGenre objects within a randomly
+        selected window of values for the `random` field.
 
-        # Setting a limit ensures that improperly distributed values
-        # for Work.random can't cause the query to return more than
-        # five times the number of records we need. If this happens,
-        # it's still bad -- we might not get records for the later
-        # lanes -- but at least the query won't run forever.
-        qu = qu.limit(target_size * len(lanes) * 5)
+        :param facets: A `FeaturedFacets` object.
 
-        return qu
+        :param target_size: Try to get approximately this many
+        items. There may be more or less; this controls the size of
+        the window and the LIMIT on the query.
+        """
+        from model import MaterializedWorkWithGenre
+        work_model = MaterializedWorkWithGenre
 
-    @classmethod
-    def _add_lane_id_field(cls, _db, qu, lanes, target_size):
```

Comment:
This method isn't needed anymore because we no longer use a single query to pull from multiple lanes.